src/Options/Applicative/Help/Style.hs

```diff
@@ -0,0 +1,124 @@
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+
+module Options.Applicative.Help.Style
+  ( SetStyle (..)
+  , ColorIntensity (..)
+  , Layer (..)
+  , ConsoleIntensity (..)
+  , Underlining (..)
+  , Italicized (..)
+  , Color (..)
+  , color
+  , bgColor
+  , colorDull
+  , bgColorDull
+  , bold
+  , underlined
+  , italicized
+  , styleToRawText
+  , defaultStyle
+  ) where
+
+import Control.Applicative
+import Data.Maybe
+import System.Console.ANSI (ConsoleIntensity (..), ColorIntensity (..), Underlining (..))
+
+import qualified System.Console.ANSI    as ANSI
+
+data SetStyle = SetStyle
+  { ansiReset             :: Bool
+  , ansiForeground        :: Maybe (ColorIntensity, Color)  -- ^ Set the foreground color, or keep the old one.
+  , ansiBackground        :: Maybe (ColorIntensity, Color)  -- ^ Set the background color, or keep the old one.
+  , ansiConsoleIntensity  :: Maybe ConsoleIntensity         -- ^ Adjust boldness
+  , ansiItalics           :: Maybe Italicized               -- ^ Adjust italics
+  , ansiUnderlining       :: Maybe Underlining              -- ^ Adjust underlining
+  } deriving (Eq, Ord, Show)
+
+instance Monoid SetStyle where
+    mempty = SetStyle False Nothing Nothing Nothing Nothing Nothing
+    mappend = (<>)
+
+defaultStyle :: SetStyle
+defaultStyle = SetStyle
+  { ansiReset             = True
+  , ansiForeground        = Nothing
+  , ansiBackground        = Nothing
+  , ansiConsoleIntensity  = Just NormalIntensity
+  , ansiItalics           = Just NoItalics
+  , ansiUnderlining       = Just NoUnderline
+  }
+
+isItalicised :: Italicized -> Bool
+isItalicised Italicized = True
+isItalicised NoItalics = False
+
+styleToRawText :: SetStyle -> String
+styleToRawText = ANSI.setSGRCode . stylesToSgrs
+  where
+    stylesToSgrs :: SetStyle -> [ANSI.SGR]
+    stylesToSgrs (SetStyle r fg bg b i u) = catMaybes
+        [ if r then Just ANSI.Reset else Nothing
+        , fmap (\(intensity, c) -> ANSI.SetColor ANSI.Foreground intensity (convertColor c)) fg
+        , fmap (\(intensity, c) -> ANSI.SetColor ANSI.Background intensity (convertColor c)) bg
+        , fmap ANSI.SetConsoleIntensity b
+        , fmap (ANSI.SetItalicized . isItalicised) i
+        , fmap ANSI.SetUnderlining u
+        ]
+
+    convertColor :: Color -> ANSI.Color
+    convertColor = \c -> case c of
```

Comment:
Use `\case`? (Since GHC 7.6)
```haskell
convertColor = \case
  Black -> ...
  ...
```

Reply:
I would prefer to keep extensions low for this library.

Reply:
Yeah, as much as I prefer using the extension I've avoided it to keep in line with existing code.